tests/Unit/Core/FeatureFlag/FeatureFlagsModifierTest.php

```diff
@@ -0,0 +1,258 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Core\FeatureFlag;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ObjectRepository;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagsModifier;
+use PrestaShopBundle\Entity\FeatureFlag;
+use Symfony\Component\Translation\TranslatorInterface;
+use TypeError;
+
+class FeatureFlagsModifierTest extends TestCase
+{
+    public function testImplementsDataConfigurationInterface()
```

Comment:
Test for behavior, not implementation. This test doesn't add value IMO

Reply:
It makes sure that, if one day DataConfigurationInterface evolves, the class will still implement it. It aims to catch interface<->class mismatch following a code change.

Reply:
I agree with @eternoendless, I see no reason to test if a class implements another one if it hasn't been created by a builder/factory or anything else. Here, you're only doing a `new myClass` and test if `myClass` correctly implements `DataConfigurationInterface`, this is something your code can test, instead of:

```php
blablafunction(FeatureFlagsModifier $ffmodifier);
```
do
``` 
blablafunction(DataConfigurationInterface $ffmodifier);
```

Reply:
@PierreRambaud I dont understand 😅  on one hand you try to persuade me to test my getters and setters, one the other hand you tell me not to test something because some unrelated class can verify this and fail.

I will _remove this test_ to make you and @eternoendless happy 🤣  but I was clearly not expecting having to defend a unit test in a CR

Reply:
It's clearly not the same. It always depends on what you want to test, here, it brings no value. 
I already saw a migration PR without tests on getter that wasn't working well.

As example:
```php
public function getSomething(): array
{
    return [
        'html' => static::TYPE_HTML,
        'text' => static::TYPE_HTML,
    ];
}
```

If none of your tests are covering this part properly, you can live with it without noticing anything for years, it costs nothing to add a tests here because you want to be sure the return values are equal. But testing an implementation of an interface can easily be done by your PHP code.

Reply:
I think I start to understand.

Basically you are saying "a test that verifies that class A implements interface Z" is easy to do, and if A does not implement Z, then it will be discovered very quick so no need to test that" while "a test that verifies that getSomething() returns valid data" is also easy to do, but if getSomething() does not return the right data, then we might not detect this for a long time, so testing it is important".

Is this your idea? "Write tests depending on whether the defect being monitored is likely to go unnoticed for a long time"?